tests/bluetooth/host/buf/bt_buf_get_evt/src/main.c

```diff
@@ -92,14 +92,17 @@ static struct net_buf_pool *get_memory_pool(bool discardable)
  */
 ZTEST(bt_buf_get_evt_default_events_returns_not_null, test_returns_not_null)
 {
-	static struct net_buf expected_buf;
+#define USER_DATA_SIZE sizeof(struct bt_buf_data)
+	struct net_buf *expected_buf = alloca(sizeof(struct net_buf) + USER_DATA_SIZE);
```

Comment:
Using `alloca` can potentially cause portability issues because it is not a standard function (not part of ISO C or even POSIX).

`alloca` or VLA should not even be necessary here since all size components are statically computable.

Reply:
Thanks, we were not aware that `alloca` is non-standard.

> `alloca` or VLA should not even be necessary here since all size components are statically computable.

Can you elaborate on this? `struct net_buf` has a flexible array member. We are not sure how to legally instantiate this object without a `malloc`-like. I suppose putting it an an union with a char-array, or just type-pun an array will likely work. But is that legal?

Reply:
Something like:

```c
char expected_buf_data[SIZE];
struct net_buf *expected_buf = (struct net_buf *)expected_buf_data;
```

> But is that legal?

Yes. Essentially, it is not any different from assigning the return value of `alloca`, which is `void *`, to `struct net_buf *`.